mqtt_subscriber/oli_broker.py

```diff
@@ -0,0 +1,47 @@
+import os
+import logging
+import traceback
+import paho.mqtt.client as mqtt
+
+
+MQTT_PORT = 1883
+MINUTES_IN_HOUR = 60
+MEASUREMENT_PERIOD_MINUTES = 15
+
+
+class MQTTConnection:
+    def __init__(self, api_client_dict):
+        self.api_clients = api_client_dict
+        self.client = mqtt.Client()
+        self.client.on_connect = self.on_connect
+        self.client.on_message = self.on_message
+
+    def on_connect(self, client, userdata, flags, rc):
+        logging.debug(f"Connected with result code {str(rc)}")
+
+        for device_name in self.api_clients:
+            client.subscribe(f"{device_name}/activeEnergy/*")
+
+    def on_message(self, client, userdata, msg):
+        logging.debug(f"{msg.topic} {str(msg.payload)}")
+
+        try:
+            device_name = msg.topic.split("/activeEnergy/")[0]
+            energy = msg.payload["value"]
+            power = energy * (MINUTES_IN_HOUR / MEASUREMENT_PERIOD_MINUTES)
```

Comment:
Is the MEASUREMENT_PERIOD_MINUTES fixed on the Oli side? 
I am asking because they could publish "forecasts" more often if they like, there is no limit here. Right?

Reply:
Yes, that's what they promised us and that's why they do not send us the measurement period even though this exists in their message structure. They only send us the energy values and the timestamp in the messages.